pkgs/development/python-modules/pympler/default.nix

```diff
@@ -13,14 +16,37 @@ buildPythonPackage rec {
     sha256 = "993f1a3599ca3f4fcd7160c7545ad06310c9e12f70174ae7ae8d4e25f6c5d3fa";
   };
 
+  patches = [
+    # Fixes a TypeError on Python 3.11
+    # (see https://github.com/pympler/pympler/issues/148)
+    (fetchpatch {
+      name = "${pname}-pull-149.patch";
+      url = "https://github.com/pympler/pympler/pull/149.patch";
+      hash = "sha256-6MK0AuhVhQkUzlk29HUh1+mSbfsVTBJ1YBtYNIFhh7U=";
+    })
+  ];
+
   nativeCheckInputs = [
     pytestCheckHook
   ];
 
+  # There is a version of bottle bundled with Pympler, but it is broken on
+  # Python 3.11. Fortunately, Pympler will preferentially import an external
+  # bottle if it is available, so we make it an explicit dependency on later
+  # Pythons.
+  propagatedBuildInputs = lib.optionals (pythonAtLeast "3.11") [
```

Comment:
Suggest doing this non-conditionally - I think in general we'd prefer it to be using our own version of `bottle` if possible.